src/ekobox/evol_pdf.py

```diff
@@ -0,0 +1,177 @@
+        theory_card,
+        operators_card,
+        len(evolved_PDF_list),
+        info_update=info_update,
+    )
+    all_member_blocks = []
+    all_blocks = []
+    for evolved_PDF in evolved_PDF_list:
+        block = genpdf.generate_block(
+            lambda pid, x, Q2, evolved_PDF=evolved_PDF: targetgrid[targetgrid.index(x)]
+            * evolved_PDF[Q2]["pdfs"][pid][targetgrid.index(x)],
+            xgrid=targetgrid,
+            Q2grid=operators_card["Q2grid"],
+            pids=br.flavor_basis_pids,
+        )
+        # all_blocks will be useful in case there will be necessity to dump many blocks
+        # for a single member
+        all_blocks.append(block)
+        all_member_blocks.append(all_blocks)
+
+    genpdf.export.dump_set(name, info, all_member_blocks)
+
+    if install:
+        genpdf.install_pdf(name)
+
+
+def gen_out(theory_card, op_card, path=None):
+    """
+    Generates EKO output from theory and operators cards and, if requested,
+    dumps it in tar format
+
+    Parameters
+    ----------
+        theory_card : dict
+            theory card
+        op_card : dict
+            operators card
+        path : str
+            path of dumped output (if "None" output is not dumped)
+
+    Returns
+    -------
+        : eko.output.Output
+            eko output
+    """
+    eko_output = eko.run_dglap(theory_card, op_card)
+    if path is not None:
+        ops_id = f"o{op_card['hash'][:6]}_t{theory_card['hash'][:6]}"
+        path = f"{path}/{ops_id}.tar"
+        eko_output.dump_tar(path)
+    return eko_output
+
+
+# TODO: add a control on the theory (but before we need to implement another kind of output which includes the theory and operator runcards)
+def ekos_product(eko_ini, eko_fin, in_place=True, force=False):
+    """
+    Returns te product of two ekos
+
+    Parameters
+    ----------
+        op_eko_ini : eko.output.Output
+            initial eko operator
+        op_eko_fin : eko.output.Output
+            final eko operator
+
+    Returns
+    -------
+        : eko.output.Output
+            eko operator
+    """
+    if eko_fin["q2_ref"] not in eko_ini["Q2grid"].keys():
+        raise ValueError(
+            "Initial Q2 of final eko operator does not match any final Q2 in the initial eko operator"
+        )
+    ope1 = eko_ini["Q2grid"][eko_fin["q2_ref"]]["operators"]
+    ope1_error = eko_ini["Q2grid"][eko_fin["q2_ref"]]["operator_errors"]
+    ope2_dict = {}
+    ope2_error_dict = {}
+    for Q2 in eko_fin["Q2grid"].keys():
+        ope2_dict[Q2] = eko_fin["Q2grid"][Q2]["operators"]
+        ope2_error_dict[Q2] = eko_fin["Q2grid"][Q2]["operator_errors"]
+    final_op_dict = {}
+    final_op_error_dict = {}
+    final_alphas_dict = {}
+    final_dict = {}
+    for Q2 in ope2_dict.keys():
+        final_op_dict[Q2] = np.einsum("ajbk,bkcl -> ajcl", ope1, ope2_dict[Q2])
+        final_op_error_dict[Q2] = np.einsum(
+            "ajbk,bkcl -> ajcl", ope1_error, ope2_error_dict[Q2]
```

Comment:
Is this correct? Or was just to have and approximation?

Reply:
I'd say it's wrong, but this integration errors are almost useless, and they are becoming a plague.
I would make them optional (is a factor 2 in memory and disk storage), and do not propagate them in non-trivial places anyhow...

Reply:
In fact it is only temporary since I did not know what we wanted to do with them. So, should I simply remove them and keep the original errors from the initial ekobox operator?

Reply:
I'd set them to zeros, better than something random

Reply:
Nope, better no errors than wrong errors. You have two operators, the errors on the product are not any of: the errors of the first, the errors on the second, the product of the errors.
If you wish they should be the sum of the products of the error on one with the value of the other:

```
(a + Da)(b + Db) = ab + bDa + aDb + DaDb ~ ab + bDa + aDb
```

Reply:
However, it's a bit weird that the product of two EKOs (that does not involve any pdf) is inside the `evol_pdf` module.

Put in a separate one. In case of doubt (if there is not any other suitable module, nor it's worth a module on its own) put in a `utils.py` module (that is the generic name we use for extra stuffs).

Reply:
They should be added in quadrature no? 
```
np.sqrt((Da*b)**2 + (Db*a)**2) 
```

Reply:
It depends if you consider them statistical or not, and actually they are not (unless you have a statistical interpretation of quadrature integration, that is not Monte Carlo).

Reply:
Should be fixed now